### PR TITLE
Adds three new deathmatch maps - Ragnarok, Lattice Battles, Species Showdown

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Adds three new deathmatch maps.

### Important nonDM Balance Changes

Cult daggers fit on belts.

Heretic blades fit on belts.

Veil shifters fit on belts.

I really don't know why these didn't, it makes inventory management rather annoying at times. It also screwed my loadouts over.

Cursed Blades fit on cult robes. They're cult equipment after all.

Bronze suits fit toy watches!!

### Ragnarok

ragnarok

A vicious battle in the jungle, between the three major religious sects: Prove your deity's might! And try not to upset any primates. Or fall into the chasm.

Going clockwise:

**Cultist Invoker**: Wielding a mirror shield, shielded robes, a sword, and some bolas, this cultist has a 'well-balanced' set of equipment to annihilate their opponents.

**Cultist Artificer**: This set harnesses the blood magicks - with spells of stunning, blood rites, and ranged hallucinations, with a wicked Cursed Blade and Veil Shifter as implement, and Berserker Robes to finish the look.

**Holy Crusader**: Nullifying most, but certainly not all, of the fearsome arsenal of the opponents with the null rod at their belt, the Crusader packs a dangerous claymore and armor to protect them from the demons.

**Rat'var Apostate**: Hey, what's that guy doing there in the dark? They don't have any magic because their god is Fucking Dead, but they're still going to show up for a token effort. Good luck!

**Heretic Scribe**: This mad soul wields an antique rifle and an assortment of other dangerous relics, with a set of magic geared towards staying far apart, picking enemies off from range and evading their attacks for the final blow.

**Heretic Warrior**: With the deadliest of Mansus Magic at their disposal, this warrior is only limited by their ability to juggle all their spells at once - don't get overwhelmed sorting your spells while an Invoker runs at you with a sword and shield and makes you cry yourself to death.

### Lattice Battles

spleef

A fresh change of pace: In this pacifist map, the only way to kill your opponents is to snip the lattices and catwalks from under them. Watch your step.

### Species Warfare

species_warfare

Prove the might of your static by duking it out with every other kind of crewmember out there. Features a messy dorms, a ticking-timebomb atmos, a rather sterile robotics, a slippery closet, a fluffy medbay, and an energetic bridge.
Mirror Shields now shatter on throw (which stuns and hurts)

## Code changes

Added two new traits, TRAIT_ACT_AS_CULTIST and TRAIT_ACT_AS_HERETIC. Added these as an OR to respective IS_X checks. Added new GET_X checks for them, which do not check the trait.

Tidied up the file those are in.

Added belt_contents() to outfits, but it dosen't work....

Added a heretic rust sister-type to rust walls and floors.

Fixed a typo in cult ascension.

## Why It's Good For The Game

These maps all aim to do something interesting and unique with DM rather than the usual deathfest and hugging of random crates.

Ragnarok allows players to practice unusual and rare magical mechanics, similar to Ragin' Mages.

Lattice Battles adds SPLEEF to the game, which I think is awesome.

Species Warfare is, I think, fun and funny. Each 'department' has incredibly chaotic and thematically-appropiate content for the species its meant to symbolize, and I look forward to the chaos that every round in it is inevitably going to have.

> Mirror Shields now shatter on throw.

I'm surprised they didn't! Since they aren't used anywhere I can do what I want with them.

## Code changes

> Added two new traits, TRAIT_ACT_AS_CULTIST and TRAIT_ACT_AS_HERETIC. Added these as an OR to respective IS_X checks. Added new GET_X checks for them, which do not check the trait.

I think this is a clever solution to the problem of 'what if I want someone who acts as the antag, but isn't?' Some procs do need the datum to modify it, so there's GET_X, but those aren't common and don't seem likely to be an issue.

> Added belt_contents() to outfits, but it dosen't work....

Help would be very much appreciated, I don't know what I'm doing wrong here. It's only used for the heretic scribe's unfathomable curio.
Some post-PR cleanup coming up.

## Changelog
:cl:
add: Added three new DM maps - Ragnarok, Lattice Battles, Species Showdown.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
As an aside, I wouldn't mind if I could delete some mostly-extraneous DM maps. There's a lot of them, and... a lot are pretty damn similar. Maint Mania and Backalley, for example, they take up real estate and one of them could probably just get the axe.

**Comment by zoomachina** [comment]
> As an aside, I wouldn't mind if I could delete some mostly-extraneous DM maps. There's a lot of them, and... a lot are pretty damn similar. Maint Mania and Backalley, for example, they take up real estate and one of them could probably just get the axe.

Of all the things you could've picked as an example...

**Comment by carlarctg** [comment]
> > As an aside, I wouldn't mind if I could delete some mostly-extraneous DM maps. There's a lot of them, and... a lot are pretty damn similar. Maint Mania and Backalley, for example, they take up real estate and one of them could probably just get the axe.
> 
> Of all the things you could've picked as an example...

Of all the things I could've picked as an example, I clearly don't know what the irony is here because otherwise I wouldn't have used those as the example.

**Comment by SmArtKar** [comment]
you proooobably shouldn't include atmos cans in the species wars map, or at least make them empty

**Comment by carlarctg** [comment]
noooooooooo

**Comment by DaCoolBoss** [comment]
What's the problem with atmos cans in death match?

**Comment by Loafin34** [comment]
i think you should give the ripley the gravity pult and tesla things, so people can play around with them.

**Comment by Loafin34** [comment]
why does the chaplain get the rusted claymore? the cult warrior has a stunning shield and a 30 damage sword, why not the guy that has anti-magic?

**Comment by carlarctg** [comment]
idc

**Comment by HoodMilk** [comment]
We got spleef in ss13

**Comment by Xackii** [comment]
> As an aside, I wouldn't mind if I could delete some mostly-extraneous DM maps. There's a lot of them, and... a lot are pretty damn similar. Maint Mania and Backalley, for example, they take up real estate and one of them could probably just get the axe.

MaintMania the best map of all times.

**Comment by Xackii** [comment]
Also you added belt item flags but don't added sprites for it

**Comment by carlarctg** [comment]
They already have them

**Comment by mc-oofert** [comment]
> As an aside, I wouldn't mind if I could delete some mostly-extraneous DM maps. There's a lot of them, and... a lot are pretty damn similar. Maint Mania and Backalley, for example, they take up real estate and one of them could probably just get the axe.

no maps are extraneous but ragin mages (and also that map with the disablers FUCK that map)

**Comment by mc-oofert** [comment]
anyhow i pray the consoles in species showdown are fake and also there is no problem with gas cans they dont cause any issues with deathmatch

**Comment by carlarctg** [comment]
they arent fake but none of them are consoles that can be meaningfully interacted with

**Comment by Loafin34** [comment]
please delete securing its the worst one, its worse than mech madness

**Comment by Eronymun** [comment]
I like back alley but yeah maint maina is a bit boring.

**Comment by mc-oofert** [comment]
> I like back alley but yeah maint maina is a bit boring.

maint mania is the og

File: code/modules/clothing/suits/costume.dm [hunk 1]
```diff
@@ -511,6 +511,7 @@
 	name = "bronze suit"
 	desc = "A big and clanky suit made of bronze that offers no protection and looks very unfashionable. Nice."
 	icon_state = "clockwork_cuirass_old"
+	allowed = list(/obj/item/toy/clockwork_watch)
```
Comment (carlarctg) [inline code/modules/clothing/suits/costume.dm:514]: undocced

File: code/__DEFINES/inventory.dm [hunk 2]
```diff
@@ -63,6 +63,8 @@
 #define ITEM_SLOT_HANDCUFFED (1<<18)
 /// Legcuff slot (bolas, beartraps)
 #define ITEM_SLOT_LEGCUFFED (1<<19)
+/// Inside of a character's BELT.........
+#define ITEM_SLOT_BELTPACK (1<<15)
```
Comment (MrMelbert) [inline code/__DEFINES/inventory.dm:67]: \>Inside of a character's BELT
\>`ITEM_SLOT_BELTPACK`

File: code/__DEFINES/antagonists.dm [hunk 3]
```diff
@@ -217,22 +217,28 @@ GLOBAL_LIST_INIT(ai_employers, list(
 /// Checks if the given mob is a traitor
 #define IS_TRAITOR(mob) (mob?.mind?.has_antag_datum(/datum/antagonist/traitor))
 
+/**
+ * Cult checks
+ */
+
 /// Checks if the given mob is a blood cultist
-#define IS_CULTIST(mob) (mob?.mind?.has_antag_datum(/datum/antagonist/cult))
+#define IS_CULTIST(mob) (mob?.mind?.has_antag_datum(/datum/antagonist/cult) || HAS_TRAIT(mob, TRAIT_ACT_AS_CULTIST))
```
Comment (MrMelbert) [inline code/__DEFINES/antagonists.dm:225]: My 2c says you could instead just have `TRAIT_CULTIST` or `TRAIT_HERETIC`, replace relevant `IS_CULTIST` / `IS_HERETIC` checks (like in sword code) with `HAS_TRAIT(TRAIT_CULTIST)` etc, and then make those antag datums give those traits

File: code/modules/antagonists/cult/blood_magic.dm [hunk 4]
```diff
@@ -433,13 +433,15 @@
...
+	var/datum/antagonist/cult/cultist = GET_CULTIST(user)
+	var/datum/team/cult/cult_team = cultist?.get_team()
 	var/effect_coef = 1
-	if(cult_team.cult_ascendent)
+	if(cult_team?.cult_ascendent)
 		effect_coef = 0.1
-	else if(cult_team.cult_risen)
+	else if(cult_team?.cult_risen)
 		effect_coef = 0.4
+	if(IS_CULTIST(user) && isnull(GET_CULTIST(user)))
```
Comment (MrMelbert) [inline code/modules/antagonists/cult/blood_magic.dm:443]:
```suggestion
	// cultists who are not in a cult
	if(IS_CULTIST(user) && isnull(cultist))
```

File: code/modules/clothing/suits/costume.dm (hunk 1 above)
Comment (MrMelbert) [inline code/modules/clothing/suits/costume.dm:514]: We lose all the normal allowed list stuff like internals tanks?

**Comment by Ghommie** [comment]
Check that there are no wallening-related issues with the 'Species Warfare' map, and perhaps the other two as well.

**Pull request merged**